tools/src/bin/tquic_client.rs

```diff
@@ -1015,9 +1016,32 @@ impl RequestSender {
         }
     }
 
-    fn send_request(&mut self, conn: &mut Connection) -> Result<()> {
+    fn get_url(&self) -> Url {
         let url = &self.option.urls[self.current_url_idx];
-        let mut request = Request::new("GET", url, &None, &self.option.dump_dir);
+        let path = url.path();
+
+        // Define the regex pattern to match %random[min:max]} and check if the pattern matches
+        let re = Regex::new(r"\%random\[(\d+):(\d+)\]").unwrap();
```

Comment:
The format of random URLs may vary depending on different use cases.

If you have such a need, it's recommended to use a shell script to generate a URL list file and modify tquic_client to support the `url_file` option to read the URL list from the file.

Reply:
The issue with the ·shell script· is that it needs to start a new tquic_client process repeatedly, with each process fetching only one file from the server. This results in significant overhead.

Reply:
You can specify [multiple URLs instead of just one](https://tquic.net/docs/getting_started/demo#client). 
```
Usage: tquic_client [OPTIONS] [URLS]...
```

Note that due to command line length limitations, if you have too many URLs, you might need to add a `url_file` option to tquic_client to read a batch of URLs from a specified file.

Reply:
Thanks. Not known if this approach will work for our case, as we need to evaluate the random access performance with 10000 concurrent clients from a total of 1 million (or possibly more) files on the QUIC web server. That's why we are adding a pattern here to directly generate random URLs in the code.